server/github_hook_handler.go

```diff
@@ -111,52 +135,186 @@ func (gh *githubHookHandler) processEvent(eventType string, deliveryID string, p
 	}
+
 	deRequest := github.CreateWorkflowDispatchEventRequest{
-		Ref:    "main",
+		Ref:    pipeline.TargetBranch,
 		Inputs: inputs,
 	}
+
 	_, err = client.Actions.CreateWorkflowDispatchEventByFileName(ctx, pipeline.Organization, pipeline.Repository, pipeline.Workflow, deRequest)
 	if err != nil {
-		log.
-			WithError(err).
+		log.WithError(err).
 			WithFields(log.Fields{
-				"installation_id": eventContext.GetInstallationID(),
-				"org":             pipeline.Organization,
-				"repo":            pipeline.Repository,
-				"workflow":        pipeline.Workflow,
-			}).
-			Error("Error occurred while triggering pipeline!")
+				"org":      pipeline.Organization,
+				"repo":     pipeline.Repository,
+				"workflow": pipeline.Workflow,
+				"branch":   pipeline.TargetBranch,
+			}).Error("Error occurred while triggering pipeline!")
 		return errors.Wrap(err, "")
 	}
 
+	if err := gh.waitForWorkflow(ctx, eventContext, pipeline); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func (gh *githubHookHandler) waitForWorkflow(ctx context.Context, eventContext model.EventContext, pipeline config.PipelineConfig) error {
+	wf, _ := gh.getCreatedPipeline(ctx, eventContext, pipeline)
+	gh.sendStatus(ctx, eventContext, pipeline, wf, statePending)
+	state, _ := gh.waitForWorkflowStatus(ctx, eventContext, pipeline, wf, 15*time.Second)
+	gh.sendStatus(ctx, eventContext, pipeline, wf, state)
+
+	return nil
+}
+
+func (gh *githubHookHandler) getCreatedPipeline(ctx context.Context, eventContext model.EventContext, pipeline config.PipelineConfig) (*github.WorkflowRun, error) {
+	log.WithFields(log.Fields{
+		"type":     "get",
+		"org":      pipeline.Organization,
+		"repo":     pipeline.Repository,
+		"workflow": pipeline.Workflow,
+	}).Info("Getting current running Workflow")
+
+	client, err := gh.ClientManager.Get(eventContext.GetInstallationID())
+	if err != nil {
+		log.WithError(err).
+			WithField("installation_id", eventContext.GetInstallationID()).Error("Cannot find installation id at cache!")
+		return nil, err
+	}
+
+	log.Info("Waiting for 5 seconds for the Workflow to settle")
+	time.Sleep(5 * time.Second)
+
+	listRequest := github.ListWorkflowRunsOptions{
+		Branch:  pipeline.TargetBranch,
+		Event:   "workflow_dispatch",
+		Created: fmt.Sprintf(">=%s", "2006-01-02"),
+		ListOptions: github.ListOptions{
+			PerPage: 1,
+		},
+	}
+
+	// It's always the last one since it's the one triggered . We will have to create a state within the webhook to keep track
+	workflowRuns, _, err := client.Actions.ListWorkflowRunsByFileName(ctx, pipeline.Organization, pipeline.Repository, pipeline.Workflow, &listRequest)
+	if err != nil {
+		return nil, err
+	}
+	log.Info(fmt.Sprintf("Found %d workflows", *workflowRuns.TotalCount))
+
+	return workflowRuns.WorkflowRuns[0], nil
+}
+
+func (gh *githubHookHandler) sendStatus(ctx context.Context, eventContext model.EventContext, pipeline config.PipelineConfig, wf *github.WorkflowRun, state string) {
```

Comment:
I think this should return an error to be handled respectively instead of logging it only.

Reply:
I will add a retry mechanism there also since we need the status to happen